### ISSUE TITLE
Crashes on OEM PSP-1001 hardware

### ISSUE THREAD

**Issue by plscks** [MottZilla/SOTN_AR#3]
I'm getting semi-random crashes on the PSP version. It is on OEM hardware
so maybe it is expected?

Model: PSP-1001
OS: 6.61 PRO-C
ISO Mode: Inferno
RAM: 32MB
CPU: 333/166

I've verified that I am using the EU version with matching SHA1/MD5, language
is set to English and I am using the save file delivered with the AR software.
No errors or unexpected things happen in the process of patching.

My initial attempts were made with Maria mode set to Nimble, complexity 6,
Dracula fight enabled, both crashed on trying to use a potion with Up + Circle.
Then I tried the Standard preset with Maria, complexity 6, both Dracula fight
enabled and disabled. Both crashed on potion use and both experienced random
crashes throughout the entrance typically while pressing a combination of
buttons such as Cross + Square to attack while jumping. My next test was with
an Alucard Nimble seed, complexity 6, Random items, and no additional options.
This crashed somewhere in the Alchemy Lab, I did not notice anything specific
that triggered it.

Maybe this is expected and has to do with the limited 32MB of RAM that the
PSP-1001 has? I'd happily answer any follow up questions you might have,
either here or on Discord (username plscks in The Long Library server and SOTN
Decomp server), try out future builds, or anything of that nature.

In addition, I'd just like to thank you for making this project, it is really
neat and much appreciated!

*edit - I'm using the latest release: Build: Jan 12 2024 - 03:29:59

**Comment by MottZilla** [comment]
Release 2024-01-17 addresses the issue with using Potions on real PSP hardware. There was an error fixed in that code. The Rando was briefly tested by me on a PSP-2001 model with no further issues encountered. The Rando *should* be compatible with original hardware but I do not have any other models to test with.

**Comment by plscks** [comment]
I've verified that two different Maria seeds function correctly when using a potion, no crashes or other issues. I've also verified that the following Alucard seeds are working correctly up through Alchemy Lab with no crashes:

- Standard, complexity 6, equipment pool variety
- Standard, complexity 6, equipment pool variety, Retain Death Skip
- Standard, complexity 6, equipment pool variety, Retain Death Skip, Maria/Richter Battle Dracula
- Nimble, complexity 6, equipment pool variety
- Nimble, complexity 6, equipment pool variety, Retain Death Skip
- Nimble, complexity 6, equipment pool variety, Retain Death Skip, Maria/Richter Battle Dracula
- Nimble, complexity 6, equipment pool variety, all options selected

I check the Maria/Richter one just in case in my original issue I'd accidentally left it selected from generating a Maria seed beforehand.

Everything is looking corrected in the newest build, thank you for looking into such a strange issue that was on rather old original hardware, correcting it, and doing it so quickly as well! I appreciate it a lot!
